modules/renter/contractor/contractor.go

```diff
@@ -133,6 +134,7 @@ func newContractor(cs consensusSet, w wallet, tp transactionPool, hdb hostDB, p
 
 		cachedRevisions: make(map[types.FileContractID]cachedRevision),
 		contracts:       make(map[types.FileContractID]modules.RenterContract),
+		downloaders:     make(map[types.FileContractID]*hostDownloader),
```

Comment:
a renew should invalidate any existing downloaders. The contractor will know both the old and new contract IDs, so it can easily loop through `downloaders` and invalidate any IDs that were renewed. This hasn't been implemented yet though.